### PR TITLE
Shell extend meta keys

### PR DESCRIPTION
**Pull request by jakub-uC**
Added following meta-keys:

- Ctrl-B - moves the cursor backward one character
- Ctrl-D - deletes the character under the cursor
- Ctrl-F - moves the cursor forward one character
- Ctrl-K - deletes from the cursor to the end of the line
- Alt-B - moves the cursor backward one word
- Alt-F - moves the cursor forward one word

### CODE REVIEW

File: subsys/shell/shell_ops.c [hunk 1]
```diff
@@ -97,6 +98,51 @@ void shell_op_cursor_move(const struct shell *shell, s16_t val)
...
+
+	return ret;
+}
+
+void shell_op_cursor_word_move(const struct shell *shell, s16_t val)
+{
+	s16_t shift;
+	s16_t sign;
+
+	if (val < 0) {
```
Comment (pizi-nordic) [inline subsys/shell/shell_ops.c]: Prototype does not match body: Regardless of the val, the movement will be the same in each direction.
What happen with val = 0?

**Review by pizi-nordic — APPROVED** [review]

**Comment by jukkar** [comment]
I wonder if user could get information about the keys somehow, for example CTRL-h or "shell -h" etc.

**Review by Mierunski — APPROVED** [review]

File: doc/subsystems/shell/shell.rst [hunk 2]
```diff
@@ -365,11 +365,19 @@ The shell module supports the following meta keys:
      - Action
    * - ctrl + a
      - Moves the cursor to the beginning of the line.
+   * - ctrl + b
```
Comment (dbkinder) [inline doc/subsystems/shell/shell.rst]: You might want to change all these to use ``:kbd:`ctrl + b` `` for a prettier display (as you used at the top of this document).

**Review by dbkinder — APPROVED** [review]
LGTM, but you might want to change the table near the end of the doc to use :kbd:`ctrl + b` for a prettier display (as you used at the top of this document).

**Pull request merged**